TypeCobol/Compiler/Nodes/Node.cs

```diff
@@ -612,29 +612,46 @@ public Node Get(string uri, int startIndex)
         public bool AcceptASTVisitor(IASTVisitor astVisitor) {
             bool continueVisit = astVisitor.BeginNode(this) && VisitNode(astVisitor);
 
-            if (continueVisit && CodeElement != null)
+            if (continueVisit)
             {
-                CodeElement.AcceptASTVisitor(astVisitor);
+                CodeElement?.AcceptASTVisitor(astVisitor);
             }
 
             if (continueVisit) {
                 //To Handle concurrent modifications during traverse.
                 //Get the array of Children that must be traverse.
-                Node[] childrenNodes = children.ToArray();
-                foreach (Node child in childrenNodes)
+                if (astVisitor.CanModifyChildrenNode)
                 {
-                    if (!continueVisit && astVisitor.IsStopVisitingChildren)
-                    {
-                        break;
-                    }
-                    continueVisit = child.AcceptASTVisitor(astVisitor);                    
+                    //Make copy of  children array if the visitor can modify the children
+                    Node[] childrenNodes = children.ToArray();
+                    continueVisit = VisitChildrens(astVisitor, true, childrenNodes);
+                }
+                else
+                {
+                    continueVisit = VisitChildrens(astVisitor, true, children.ToArray());
```

Comment:
I think the `ToArray()` should be removed.

Reply:
`ToArray()` can be removed if the VisitChildren signature is modified to `VisitChildrens(IASTVisitor astVisitor, bool continueVisit, IEnumerable<Node> childrenNodes)` from `VisitChildrens(IASTVisitor astVisitor, bool continueVisit, Node[] childrenNodes)`